### PR TITLE
Di   22884 create menu line menu and buttons

### PR DESCRIPTION
**Pull request by michaelatias**

### CODE REVIEW

File: client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+    getLineMenu(){
+        if(this.listContainer.Layout?.LineMenu){
+            return {
+                get: async (data: PepSelectionData) => {
+                    return this.listContainer.Layout.LineMenu.Blocks.filter(block => !block.Hidden)
+                    .map(block => {
+                        return {
+                            title: block.Title,
+                            handler: async (selectedRows) => this.lineMenuSubject.next({key: block.Key, data: selectedRows})
```
Comment (roiaharon-pepperi) [inline client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts:90]: suppose we have multiple components that registered to the lineMenuSubject, how will it work? you'll emit only the first one

File: client-side/src/app/resource-list-0.9/resource-list/resource-list.component.ts [hunk 2]
```diff
@@ -1,25 +1,73 @@
 import { Component, OnInit } from '@angular/core';
 import { ClientEventsService } from '../services/client-events.service';
-import { ViewBlocksAdapter } from '../helpers/view-blocks-adapter';
 import { ListContainer } from 'shared';
+import { IPepGenericListDataSource, IPepGenericListInitData, IPepGenericListParams } from '@pepperi-addons/ngx-composite-lib/generic-list';
+import { GenericListAdapter } from '../helpers/generic-list-adapter';
+import { GenericListAdapterResult, SmartSearchInput } from '../metadata';
+import { PepMenuItem } from '@pepperi-addons/ngx-lib/menu';
+import { GVButton } from 'src/app/generic-viewer/generic-viewer.model';
```
Comment (roiaharon-pepperi) [inline client-side/src/app/resource-list-0.9/resource-list/resource-list.component.ts:8]: remove this type, you can't reference to anything outside of resource list 0.9 folder,

File: client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import { PepButton } from "@pepperi-addons/ngx-lib/button";
+import { GVButton } from "src/app/generic-viewer/generic-viewer.model";
+import { PepAddonService, PepStyleStateType, PepStyleType } from "@pepperi-addons/ngx-lib";
+import { title } from "process";
+import { ClientEventsService } from "../services/client-events.service";
+import { Subject, async } from "rxjs";
+import { PepSelectionData } from "@pepperi-addons/ngx-lib/list";
+
+export class GenericListAdapter {
+    constructor(private listContainer: Partial<ListContainer>, private clientEventService: ClientEventsService, private lineMenuSubject: Subject<{key: string, data?: any}>){
```
Comment (roiaharon-pepperi) [inline client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts:16]: why do you need the clientEventService here? the list adapter is not the one emitting events

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts (hunk 1 above)
Comment (michaelatias) [reply client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts:90]: will be changed on the future

File: client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts [hunk 4]
```diff
@@ -1,19 +1,15 @@
...
-import { ViewBlocksAdapter } from "./view-blocks-adapter";
+import { ListContainer, ListMenuBlock } from "shared";
+import { GridViewBlockAdapter } from "./view-blocks-adapter";
 import { DataSource } from "./data-source";
-import { MenuDataView, SchemeFieldType } from "@pepperi-addons/papi-sdk";
-import { GenericListAdapterResult, SmartSearchField, SmartSearchInput } from "../metadata";
+import { GenericListAdapterResult, SmartSearchInput } from "../metadata";
 import { PepMenuItem } from "@pepperi-addons/ngx-lib/menu";
-import { PepButton } from "@pepperi-addons/ngx-lib/button";
 import { GVButton } from "src/app/generic-viewer/generic-viewer.model";
```
Comment (roiaharon-pepperi) [inline client-side/src/app/resource-list-0.9/helpers/generic-list-adapter.ts:6]: update this import to point ./metadata

File: client-side/src/app/resource-list-0.9/resource-list/resource-list.module.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ import { PepButtonModule } from '@pepperi-addons/ngx-lib/button';
 @NgModule({
   declarations: [
     ResourceListComponent,
+    ListUIComponent,
```
Comment (roiaharon-pepperi) [inline client-side/src/app/resource-list-0.9/resource-list/resource-list.module.ts:15]: why have you added it?

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

**Review by roiaharon-pepperi — APPROVED** [review]

**Pull request merged**